src/result.ts

```diff
@@ -61,10 +61,15 @@ export namespace Result {
 
 export type Result<T, E> = Ok<T, E> | Err<T, E>
 
-export const ok = <T, E = never>(value: T): Ok<T, E> => new Ok(value)
+export function ok<T, E = never>(value: T): Ok<T, E>
+export function ok<T extends void = void, E = never>(value: void): Ok<void, E>
+export function ok<T, E = never>(value: T): Ok<T, E> {
+  return new Ok(value)
+}
 
 export function err<T = never, E extends string = string>(err: E): Err<T, E>
 export function err<T = never, E = unknown>(err: E): Err<T, E>
+export function err<T = never, E extends void = void>(err: void): Err<T, void>
```

Comment:
Is there really any valid use-case for void errors?

Reply:
I've never used one personally. Perhaps someone is using it as a generic "failure" while making a quick and dirty prototype.

I included the `err` overload for completeness, not for any specific use case.

I think it would be more surprising for the consumer if ok/err didn't work the same way. Do you see any downside to this?

Reply:
In my humble opinion, this is a case of YAGNI.
It feels _wrong_ to be able to return an error which is void.

Reply:
My 2c... 

I don't disagree with you @Papooch . I don't really see a use case for it either. Doesn't mean there isn't one.

However, I will say I think there's an overYAGNIation in libraries in general. So many times I've come across a feature I've wanted and it hasn't been implemented because of YAGNI. No human being can think of all the possible use cases, and in this case, it's not like adding that overload has any drawbacks.

At the end of the day though, I don't really mind either way - I'd just like to see the void Ok implemented! So whatever makes everyone else happy to expedite that process is fine by me! :)

Reply:
That's ok for me. 
It looks like that Rust's Err accepts Unit type